### PR TITLE
Keep micro commits

### PR DESCRIPTION
**Pull request by gregorriegler**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,8 @@ Basic Commands:
   next               handover changes in wip branch to next person
   done               squashes all changes in wip branch to index in base branch
   reset              removes local and remote wip branch
+  squash-wip		 combines wip commits in wip branch with subsequent manual commits to leave only manual commits
```
Comment (simonharrer) [inline README.md:88]: Indent seems off for the description

File: README.md [hunk 2]
```diff
@@ -99,6 +101,9 @@ Basic Commands(Options):
     [--squash]                           Squash commits from wip branch
   reset
     [--branch|-b <branch-postfix>]       Set wip branch to 'mob/<base-branch>/<branch-postfix>'
+  squash-wip		 
+	[--git-editor]						 Not intended for manual use. Used as a non-interactive editor (GIT_EDITOR) for git.
```
Comment (simonharrer) [inline README.md:105]: indent seems off for the description

File: mob.go [hunk 3]
```diff
@@ -839,6 +851,8 @@ Basic Commands:
   next               handover changes in wip branch to next person
   done               squashes all changes in wip branch to index in base branch
   reset              removes local and remote wip branch
+  squash-wip		 combines wip commits in wip branch with subsequent manual commits to leave only manual commits 
+					 ! Works only if all wip commits have the same wip message !
```
Comment (simonharrer) [inline mob.go:855]: indent seems off

File: mob.go [hunk 4]
```diff
@@ -853,6 +867,9 @@ Basic Commands(Options):
     [--squash]                           Squash commits from wip branch
   reset 
     [--branch|-b <branch-postfix>]       Set wip branch to 'mob/<base-branch>/<branch-postfix>'
+  squash-wip		 
+	[--git-editor]						 Not intended for manual use. Used as a non-interactive editor (GIT_EDITOR) for git.
```
Comment (simonharrer) [inline mob.go:871]: indent seems off

File: squash_wip_test.go [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+func TestMarkSquashWip_wipCommitFollowedByManualCommit(t *testing.T) {
+	configuration = getDefaultConfiguration()
+	input := "pick 01a9a31 " + configuration.WipCommitMessage + "\n" +
+		"pick c51a56d manual commit\n" +
+		"\n" +
+		"# Rebase ..."
+	expected := "pick 01a9a31 " + configuration.WipCommitMessage + "\n" +
+		"squash c51a56d manual commit\n" +
+		"\n" +
+		"# Rebase ..."
```
Comment (simonharrer) [inline squash_wip_test.go:176]: please use multi-line strings for these cases. see https://yourbasic.org/golang/multiline-string/ for an example

File: README.md (hunk 1 above)
Comment (gregorriegler) [reply README.md]: fixed

File: mob.go (hunk 3 above)
Comment (gregorriegler) [reply mob.go]: fixed

File: squash_wip_test.go (hunk 5 above)
Comment (gregorriegler) [reply squash_wip_test.go]: did refactor to multiline strings

**Comment by gregorriegler** [comment]
@simonharrer i was able to test it on windows. it successfully combined a wip commit and a manual commit.
however the windows build didn't work, but that's another story and another pr.

File: README.md (hunk 2 above)
Comment (gregorriegler) [reply README.md]: fixed

**Comment by gregorriegler** [comment]
@simonharrer do you perhaps have a mac to test this?

File: mob.go [hunk 6]
```diff
@@ -874,6 +888,9 @@ Basic Commands(Options):
     [--squash]                           Squash commits from wip branch
   reset 
     [--branch|-b <branch-postfix>]       Set wip branch to 'mob/<base-branch>/<branch-postfix>'
+  squash-wip
```
Comment (simonharrer) [inline mob.go:891]: would like this to be in a separate group. Maybe as an experimental feature first until it’s battle tested?

**Comment by simonharrer** [comment]
> @simonharrer do you perhaps have a mac to test this?

you mean to run the tests on osx? Or manual testing?

**Comment by gregorriegler** [comment]
> > @simonharrer do you perhaps have a mac to test this?
> 
> you mean to run the tests on osx? Or manual testing?

yes, both.

File: mob.go (hunk 6 above)
Comment (gregorriegler) [reply mob.go:891]: like so ?

**Comment by gregorriegler** [comment]
do you have any concerns?

File: mob.go (hunk 6 above)
Comment (simonharrer) [reply mob.go:891]: It should no longer be here and part of the basic commands, only in the experimental section.

**Comment by simonharrer** [comment]
Tested it on OS X. Works fine. Some more comments:

```
➜  mob git:(mob/keep-micro-commits) mob squash-wip
 > Make sure the final commit is a manual commit before squashing
```

This should be an ERROR because the user expected something to happen and it did not. I would rephrase that as well: 

```
ERROR failed to squash wip commits
ERROR last commit must be a manual commit
ERROR create a manual commit with a commit message to fix this:

     git commit --allow-empty

```

**Comment by simonharrer** [comment]
After a successful execution of `mob squash-wip-commits` we need to print to the user that history has been rewritten and guide them towards the mob done command. Maybe show something like: 

```
> to put the changes into the base branch call 

    mob done --no-squash

```

**Comment by simonharrer** [comment]
You can use `sayError` and `sayTodo` for the output markup.

File: mob_test.go [hunk 7]
```diff
@@ -982,12 +994,14 @@ func createFileAndCommitIt(t *testing.T, filename string, content string, commit
 	git("commit", "-m", commitMessage)
 }
 
-func createFile(t *testing.T, filename string, content string) {
+func createFile(t *testing.T, filename string, content string) (path string) {
 	d1 := []byte(content)
```
Comment (simonharrer) [inline mob_test.go:998]: renaming this variable to something more expressive would be nice (although it's not part of this PR, but still - just have seen that)

File: squash_wip.go [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+func squashWip(configuration Configuration) {
+	if endsWithWipCommit(configuration) {
+		sayInfo("Make sure the final commit is a manual commit before squashing")
+		exit(1)
+		return
+	}
+
+	currentBaseBranch, currentWipBranch := determineBranches(gitCurrentBranch(), gitBranches(), configuration)
+	if gitCurrentBranch() != currentWipBranch {
+		sayInfo("Make sure you are on the wip-branch before running quash-wip")
```
Comment (simonharrer) [inline squash_wip.go:21]: In this case, one isn't mob programming. Try to use the same logic as in other methods when one should be mob programming but isn't.

File: mob_test.go (hunk 7 above)
Comment (gregorriegler) [reply mob_test.go]: > Tested it on OS X. Works fine. Some more comments:
> 
> ```
> ➜  mob git:(mob/keep-micro-commits) mob squash-wip
>  > Make sure the final commit is a manual commit before squashing
> ```
> 
> This should be an ERROR because the user expected something to happen and it did not. I would rephrase that as well:
> 
> ```
> ERROR failed to squash wip commits
> ERROR last commit must be a manual commit
> ERROR create a manual commit with a commit message to fix this:
> 
>      git commit --allow-empty
> ```

Thank you for testing. Also, I like your remarks regarding error and guidance.
Improved those messages, to even show the resulting commits. Really like the sayTodo.
Feels nice now.
Also found a bug in the process. An empty commit broke squash-wip. I fixed that.

File: squash_wip.go (hunk 8 above)
Comment (gregorriegler) [reply squash_wip.go]: good catch, did that.
